rfcs/0199-basic-config.md

```diff
@@ -0,0 +1,91 @@
+
+This configuration file could be set via an environment variable such as `MNEMOS_CONFIG=/path/to/beepy.toml`, which would be loaded by a `build.rs` script, and set by `just`.
+
+In this way, customized targets can be made by creating a new configuration file, rather than an entire cargo project.
+
+## Compile Time vs Runtime Configuration
+
+At the moment, most configuration will be done at compile time, with configuration values being "baked into" the binary.
+
+However, we should ensure that all configuration values also implement `serde` traits, so that platforms that CAN load configuration at runtime (from an SD Card or other external sources) are able to.
+
+Platforms that can load configuration at runtime may need to provide a "two stage" initialization process: using enough of the "baked in" configuration to initialize the SD Card or other external source, then continuing with the initialization process after merging or replacing the baked in data with the loaded data.
+
+## Inspiration
+
+This approach is largely inspired by [ICU4X]'s [Data Provider] model, which aims to "bake in" some common assets, but also allow for loading additional assets upon request.
+
+[ICU4X]: https://github.com/unicode-org/icu4x
+[Data Provider]: https://github.com/unicode-org/icu4x/blob/main/docs/design/data_pipeline.md
+
+## "Tomorrows Problems"
+
+This initial approach avoids many complexities by ignoring them completely. Some of these complexities are described below:
+
+### Compile time feature enablement
+
+At some point, we may have many drivers, some of which that are not relevant or possible to support on some platforms. Since this system will support runtime configuration, the compiler will still need to compile them into the kernel image, even though there is no chance of them being used.
+
+This approach does not address compile time gating of features to prevent certain capabilities from being compiled in at all.
+
+While this may lead to some "code bloat", we are not at a stage where this is an immediate concern. Once we are, our configuration approach may need to be revised.
+
+### Extensibility
+
+Similarly as new features are added that COULD be supported on all platforms, we will need to add them manually to the configuration items of each crate.
+
+For example: if we have SPI drivers in many platforms, and write a bunch of drivers for external components connected over SPI, we will need to add configuration for each driver to all platforms.
+
+This might be mitigated by grouping these into a common "all spi drivers" configuration type that is included "by composition" into all platform crates, but at some point this will cause friction and/or maintenance burden.
+
+We may want to revisit how to support that in the future.
+
+### "demand based" configuration
```

Comment:
maybe also worth mentioning runtime device detection/plug-n-play as out of scope here?